src/frontend/src/catalog/table_catalog.rs

```diff
@@ -273,6 +276,50 @@ impl TableVersion {
     }
 }
 
+impl TableCatalog {
+    /// Returns the SQL definition when the table was created, purified with best effort
+    /// if it's a table.
+    pub fn create_sql_purified(&self) -> String {
+        self.create_sql_ast_purified()
+            .map(|stmt| stmt.to_string())
+            .unwrap_or_else(|_| self.create_sql())
+    }
```

Comment:
Should we still return the result here? Not sure if we can guarantee this for all exist tables created previously.

Reply:
By falling back to `self.create_sql()` (i.e. `self.definition`) we ensure that the behavior gets strictly better compared to prior to this PR. 🤔 Not sure what you mean by "guarantee this"?